### PR TITLE
Added transaction status in TransactionRepository

### LINKED ISSUE #374: Facilitator should not send (retry to send) the transactions if the estimate gas fails.
If a transaction fails, there is no logic to stop the retry. The facilitator will keep retrying it. This can be critical when there will be competing facilitators. The fix can be we can add a status or retry column in the transaction table.

### PR DESCRIPTION
**Pull request by abhayks1**
Fixes #374

### CODE REVIEW

File: src/m1_facilitator/models/Transaction.ts [hunk 1]
```diff
@@ -15,22 +15,34 @@
 import BigNumber from 'bignumber.js';
 import Comparable from '../../common/observer/Comparable';
 
+/**
+ * Type of statuses.
+ */
+export enum Status {
```
Comment (deepesh-kn) [inline src/m1_facilitator/models/Transaction.ts]: TransactionStatus

File: src/m1_facilitator/models/Transaction.ts [hunk 2]
```diff
@@ -15,22 +15,34 @@
...
+  Pending = 'pending',
+  Success = 'success',
```
Comment (deepesh-kn) [inline src/m1_facilitator/models/Transaction.ts]:
```suggestion
Sent = 'sent',
```

File: src/m1_facilitator/models/Transaction.ts [hunk 3]
```diff
@@ -15,22 +15,34 @@
 import BigNumber from 'bignumber.js';
 import Comparable from '../../common/observer/Comparable';
 
+/**
+ * Type of statuses.
```
Comment (deepesh-kn) [inline src/m1_facilitator/models/Transaction.ts]: Please add document, when these status are updated.
What is pending, sent, and failed

File: src/m1_facilitator/models/Transaction.ts [hunk 4]
```diff
@@ -15,22 +15,34 @@
...
-  /** Raw transaction object */
+  /** Raw transaction object. */
   public readonly encodedData: string;
 
-  /** Gas price value at which transaction was sent */
+  /** Gas price value at which transaction was sent. */
   public readonly gasPrice: BigNumber;
 
+  /** Transaction status. */
+  public status: Status;
```
Comment (deepesh-kn) [inline src/m1_facilitator/models/Transaction.ts]:
```suggestion
  public transactionStatus: TransactionStatus;
```

**Review by deepesh-kn — CHANGES_REQUESTED** [review]

File: src/m1_facilitator/models/Transaction.ts (hunk 1 above)
Comment (abhayks1) [reply src/m1_facilitator/models/Transaction.ts]: Done!

File: src/m1_facilitator/models/Transaction.ts (hunk 2 above)
Comment (abhayks1) [reply src/m1_facilitator/models/Transaction.ts]: Done!

File: src/m1_facilitator/models/Transaction.ts (hunk 3 above)
Comment (abhayks1) [reply src/m1_facilitator/models/Transaction.ts]: Done!

File: src/m1_facilitator/models/Transaction.ts (hunk 4 above)
Comment (abhayks1) [reply src/m1_facilitator/models/Transaction.ts]: Done!

**Review by deepesh-kn — APPROVED** [review]

**Pull request merged**